src/apple/system_profile.rs

```diff
@@ -5,7 +5,11 @@ use thiserror::Error;
 #[derive(Debug, Error)]
 pub enum Error {
     #[error(transparent)]
-    SystemProfilerFailed(#[from] util::RunAndSearchError),
+    SystemProfilerCommandFailed(#[from] util::RunAndSearchError),
+    #[error("\tError: {0:?}\nFailed to parse the output of system_profiler. This is a bug - please report it!")]
```

Comment:
I'd prefer we got back to just having the original `SystemProfilerFailed` variant, instead of the new `SystemProfilerCommandFailed`/`SystemProfilerRegexFailed` split. After all, `util::RunAndSearchError` specifically exists to avoid needing to split them up.